In [1]:
#import pandas and numpy
import pandas as pd
import numpy as np

In [70]:
#visualization
# we are using the inline backend
%matplotlib inline 

import matplotlib as mpl
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
#path of data
data1 = "registered-applicants.csv"
data2 = "enrolled-applicants.csv"

#dataframe
reg_df = pd.read_csv(data1)
reg_df.head(5)
    

Unnamed: 0 FIRST NAME LAST NAME        SOURCE                   COURSE  \
0        2373      Brian       May  Social Media   Python For DataScience   
1         758       Leon    Melvin  Social Media    Product Design(UI/UX)   
2        2287       Juan    Harris      Mr Simps    Product Design(UI/UX)   
3        2480    Gilbert    Denson  Social Media  Figma Design to Webflow   
4        1359      Larry  Williams       WEBSITE   Python For DataScience   

     TRACK GENDER  
0  Weekend   Male  
1  Weekend   Male  
2  Weekday   Male  
3  Weekend   Male  
4  Weekday   Male

In [3]:
enr_df = pd.read_csv(data2)
enr_df.head(5)

Unnamed: 0 FIRST NAME LAST NAME        SOURCE                   COURSE  \
0        1120     Robert    Mccain       WEBSITE               Full Stack   
1        2557     Russel    Reagin      Mr Simps               Full Stack   
2        1682      Barry   Dickson       WEBSITE               Full Stack   
3        1961      Allen  Williams      Mr Simps                      NaN   
4         295      Brian     Bryon  Social Media  Python For Data Science   

     TRACK GENDER  
0  Weekday   Male  
1  Weekday   Male  
2  Weekday   Male  
3  Weekday   Male  
4  Weekend   Male

In [4]:
enr_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1766 entries, 0 to 1765
Data columns (total 7 columns):
Unnamed: 0    1766 non-null int64
FIRST NAME    1766 non-null object
LAST NAME     1766 non-null object
SOURCE        1754 non-null object
COURSE        1741 non-null object
TRACK         1766 non-null object
GENDER        1766 non-null object
dtypes: int64(1), object(6)
memory usage: 96.7+ KB


In [5]:
enr_df.describe()

Unnamed: 0
count  1766.000000
mean   1637.131937
std     962.121444
min       2.000000
25%     822.500000
50%    1609.000000
75%    2434.750000
max    3676.000000

In [6]:
enr_df.dtypes

Unnamed: 0     int64
FIRST NAME    object
LAST NAME     object
SOURCE        object
COURSE        object
TRACK         object
GENDER        object
dtype: object

In [7]:
enr_df.isnull().sum()

Unnamed: 0     0
FIRST NAME     0
LAST NAME      0
SOURCE        12
COURSE        25
TRACK          0
GENDER         0
dtype: int64

In [8]:
enr_df.columns

Index(['Unnamed: 0', 'FIRST NAME', 'LAST NAME', 'SOURCE', 'COURSE', 'TRACK',
       'GENDER'],
      dtype='object')

In [9]:
#rename unnamed column
enr_df = enr_df.rename(columns={'Unnamed: 0':'ID'})

In [10]:
enr_df.head(5)

ID FIRST NAME LAST NAME        SOURCE                   COURSE    TRACK  \
0  1120     Robert    Mccain       WEBSITE               Full Stack  Weekday   
1  2557     Russel    Reagin      Mr Simps               Full Stack  Weekday   
2  1682      Barry   Dickson       WEBSITE               Full Stack  Weekday   
3  1961      Allen  Williams      Mr Simps                      NaN  Weekday   
4   295      Brian     Bryon  Social Media  Python For Data Science  Weekend   

  GENDER  
0   Male  
1   Male  
2   Male  
3   Male  
4   Male

In [11]:
#check for any duplicate rows
duplicated_df = enr_df[enr_df.duplicated(keep='last')]
duplicated_df

Empty DataFrame
Columns: [ID, FIRST NAME, LAST NAME, SOURCE, COURSE, TRACK, GENDER]
Index: []

In [12]:
enr_df.groupby(["GENDER"]).count()

ID  FIRST NAME  LAST NAME  SOURCE  COURSE  TRACK
GENDER                                                    
Female   142         142        142     142     142    142
Male    1611        1611       1611    1599    1586   1611
female     1           1          1       1       1      1
male      12          12         12      12      12     12

In [13]:
## change the name of rows male and female
enr_df.loc[(enr_df.GENDER == "male")] = "Male"
enr_df.loc[(enr_df.GENDER == "female")] = "Female"
    

In [14]:
#group by gender and count
enr_df.groupby(["GENDER"]).count()

ID  FIRST NAME  LAST NAME  SOURCE  COURSE  TRACK
GENDER                                                    
Female   143         143        143     143     143    143
Male    1623        1623       1623    1611    1598   1623

In [15]:
reg_df.head(10)

Unnamed: 0 FIRST NAME LAST NAME        SOURCE  \
0        2373      Brian       May  Social Media   
1         758       Leon    Melvin  Social Media   
2        2287       Juan    Harris      Mr Simps   
3        2480    Gilbert    Denson  Social Media   
4        1359      Larry  Williams       WEBSITE   
5        3516       Lori      Ross  Social Media   
6        2617    Charles    Weston  Social Media   
7        1924    Michael      Pace  Social Media   
8        2070    Micheal      Hood       WEBSITE   
9        1485     Lonnie   Uselman      Mr Simps   

                         COURSE    TRACK  GENDER  
0        Python For DataScience  Weekend    Male  
1         Product Design(UI/UX)  Weekend    Male  
2         Product Design(UI/UX)  Weekday    Male  
3       Figma Design to Webflow  Weekend    Male  
4        Python For DataScience  Weekday    Male  
5            Product Management  Weekday  Female  
6       Figma Design to Webflow  Weekday    Male  
7  Product Design(UI/UX) ONLINE  Weekday    Male  
8  Product Design(UI/UX) ONLINE  Weekday    Male  
9                 CyberSecurity   ONLINE    Male

In [16]:
reg_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3676 entries, 0 to 3675
Data columns (total 7 columns):
Unnamed: 0    3676 non-null int64
FIRST NAME    3676 non-null object
LAST NAME     3676 non-null object
SOURCE        3650 non-null object
COURSE        3651 non-null object
TRACK         3676 non-null object
GENDER        3620 non-null object
dtypes: int64(1), object(6)
memory usage: 201.1+ KB


In [17]:
reg_df.describe()

Unnamed: 0
count  3676.000000
mean   1840.485310
std    1061.339373
min       1.000000
25%     921.750000
50%    1840.500000
75%    2759.250000
max    3678.000000

In [18]:
reg_df.columns

Index(['Unnamed: 0', 'FIRST NAME', 'LAST NAME', 'SOURCE', 'COURSE', 'TRACK',
       'GENDER'],
      dtype='object')

In [19]:
#rename unnamed column
reg_df = reg_df.rename(columns={'Unnamed: 0':'ID'})

In [20]:
reg_df.head(10)

ID FIRST NAME LAST NAME        SOURCE                        COURSE  \
0  2373      Brian       May  Social Media        Python For DataScience   
1   758       Leon    Melvin  Social Media         Product Design(UI/UX)   
2  2287       Juan    Harris      Mr Simps         Product Design(UI/UX)   
3  2480    Gilbert    Denson  Social Media       Figma Design to Webflow   
4  1359      Larry  Williams       WEBSITE        Python For DataScience   
5  3516       Lori      Ross  Social Media            Product Management   
6  2617    Charles    Weston  Social Media       Figma Design to Webflow   
7  1924    Michael      Pace  Social Media  Product Design(UI/UX) ONLINE   
8  2070    Micheal      Hood       WEBSITE  Product Design(UI/UX) ONLINE   
9  1485     Lonnie   Uselman      Mr Simps                 CyberSecurity   

     TRACK  GENDER  
0  Weekend    Male  
1  Weekend    Male  
2  Weekday    Male  
3  Weekend    Male  
4  Weekday    Male  
5  Weekday  Female  
6  Weekday    Male  
7  Weekday    Male  
8  Weekday    Male  
9   ONLINE    Male

In [21]:
reg_df.dtypes

ID             int64
FIRST NAME    object
LAST NAME     object
SOURCE        object
COURSE        object
TRACK         object
GENDER        object
dtype: object

In [22]:
reg_df.isnull().sum()

ID             0
FIRST NAME     0
LAST NAME      0
SOURCE        26
COURSE        25
TRACK          0
GENDER        56
dtype: int64

In [23]:
#group by gender and count
reg_df.groupby(["GENDER"]).count()

ID  FIRST NAME  LAST NAME  SOURCE  COURSE  TRACK
GENDER                                                    
Female   713         713        713     713     713    713
Male    2883        2883       2883    2857    2858   2883
female     4           4          4       4       4      4
male      20          20         20      20      20     20

In [28]:
#check rows that has na
df1 = reg_df[reg_df.isna().any(axis=1)]
df1

ID   FIRST NAME   LAST NAME        SOURCE  \
62     192       Robert      Parker  Social Media   
124   1877       Joseph  Macdougall       WEBSITE   
190   2186      Vicente      Elkins       WEBSITE   
264   1917      Charles    Mcknight  Social Media   
278   1100       George        Wood  Social Media   
328   1087       Leslie      Montes      Mr Simps   
350    636         Paul       Young       WEBSITE   
392   1035         Nick        Lynn  Social Media   
427    571         Reed    Brookins  Social Media   
464   1679      Matthew         Joy  Social Media   
500    439         Fred       Stone  Social Media   
533   1534       Joseph       Reyes      Mr Simps   
575   2002        Harry       Ashby       WEBSITE   
598   2602       Walter      Waters  Social Media   
648    128        Larry        Crum  Social Media   
661   2895        Garry     Zombory       WEBSITE   
726   1941       Joshua       Kibby  Social Media   
731   2930       Milton       Toddy      Mr Simps   
790   1244         Juan      Oakley  Social Media   
794   2338        Frank     Bilbrey      Mr Simps   
814    785        James      Spring           NaN   
857   2714          Jan      Benson      Mr Simps   
867   1056      Jeffrey     Gravely  Social Media   
922   2438         Kyle       Hoard  Social Media   
926   1847        David      Rivera           NaN   
940    848       Vernon  Villarreal  Social Media   
986   1459        Alvin    Manzione      Mr Simps   
1012  1595         Todd        Wade  Social Media   
1036  1356       Gerald     Mcnally           NaN   
1052  1448         Jose      Pettit  Social Media   
...    ...          ...         ...           ...   
2435    26        Scott      Holmes      Mr Simps   
2446   234        Craig        Rios  Social Media   
2487  2074     Lawrence       Gomez           NaN   
2503  1434         John     Simpson       WEBSITE   
2574  1114        Brian      Cordes      Mr Simps   
2606  1645       Rodney        Hill           NaN   
2642  1659       Leland      Zeiler  Social Media   
2708  2729     Benjamin     Hubbard      Mr Simps   
2715  2590         John         Via           NaN   
2772  2040      Timothy        Ross  Social Media   
2826  1425        James        Pyle           NaN   
2838   773        Derek    Mitchell      Mr Simps   
2901  1440  Christopher      Miller  Social Media   
2935   229      Esteban     Mandell           NaN   
2966  1584       Travis      Garcia  Social Media   
3028  1950       Rodney     Hazlett       WEBSITE   
3040  2371        James       Lamar           NaN   
3094  1242       Donald         Pal  Social Media   
3154  2227         Gary        Bass           NaN   
3160  1754          Tom   Barnfield       WEBSITE   
3226    80        Louis     Merritt  Social Media   
3264   317      Quentin      Duncan           NaN   
3291   235         Eric     Gramble      Mr Simps   
3353   474       Thomas      Faines      Mr Simps   
3372   781       Carlos        Gunn           NaN   
3417  2599        Frank       Doyle       WEBSITE   
3479   786         Dean        Reed           NaN   
3485  1262        Leroy    Truelove  Social Media   
3559   566         Jose    Mckinley       WEBSITE   
3600   115       Robert     Dunmire           NaN   

                            COURSE    TRACK GENDER  
62                        Frontend   ONLINE    NaN  
124         Python For DataScience  Weekday    NaN  
190                     Full Stack  Weekday    NaN  
264                       Frontend  Weekday    NaN  
278                            NaN  Weekday   Male  
328                     Full Stack  Weekend    NaN  
350                            NaN  Weekend   Male  
392                     Full Stack  Weekday    NaN  
427                            NaN   ONLINE   Male  
464                      FullStack  Weekend    NaN  
500                            NaN  Weekend   Male  
533     Backend With Python Django  Weekday    NaN  
575                            N

In [29]:
duplicated_df2 = reg_df[reg_df.duplicated()]
duplicated_df2

Empty DataFrame
Columns: [ID, FIRST NAME, LAST NAME, SOURCE, COURSE, TRACK, GENDER]
Index: []

In [30]:
#change male and female to Male and Female
reg_df.loc[(reg_df.GENDER == "female")] = "Female"

In [31]:
reg_df.loc[(reg_df.GENDER == "male")] = "Male"

In [32]:
#group by gender and count
reg_df.groupby(["GENDER"]).count()

ID  FIRST NAME  LAST NAME  SOURCE  COURSE  TRACK
GENDER                                                    
Female   717         717        717     717     717    717
Male    2903        2903       2903    2877    2878   2903

In [33]:
#missing genders
reg_df.GENDER.isnull().sum()

56

In [38]:
#change gender to categorical 
reg_df.GENDER = reg_df.GENDER.astype("category")

reg_df.dtypes

ID              object
FIRST NAME      object
LAST NAME       object
SOURCE          object
COURSE          object
TRACK           object
GENDER        category
dtype: object

In [87]:
#lets set it as No_gender,
##gender is complicated in 2021
##add a new category and fill nan with new category
reg_df.GENDER = reg_df.GENDER.cat.add_categories("No_gender")
reg_df.GENDER.fillna("No_gender")

In [40]:
#group by gender and count
reg_df.groupby(["GENDER"]).count()

ID  FIRST NAME  LAST NAME  SOURCE  COURSE  TRACK
GENDER                                                       
Female      717         717        717     717     717    717
Male       2903        2903       2903    2877    2878   2903
No_gender     0           0          0       0       0      0

In [46]:
reg_df.GENDER.isnull().sum()

56

In [49]:
#replace nan with new category (no_gender)
reg_df["GENDER"].replace(np.nan,"No_gender", inplace = True)

In [50]:
reg_df.GENDER.isnull().sum()

0

In [54]:
#group by gender and count
reg_df.groupby(["GENDER"]).count()

ID  FIRST NAME  LAST NAME  SOURCE  COURSE  TRACK
GENDER                                                       
Female      717         717        717     717     717    717
Male       2903        2903       2903    2877    2878   2903
No_gender    56          56         56      56      56     56

In [55]:
enr_df.head(5)

ID FIRST NAME LAST NAME        SOURCE                   COURSE    TRACK  \
0  1120     Robert    Mccain       WEBSITE               Full Stack  Weekday   
1  2557     Russel    Reagin      Mr Simps               Full Stack  Weekday   
2  1682      Barry   Dickson       WEBSITE               Full Stack  Weekday   
3  1961      Allen  Williams      Mr Simps                      NaN  Weekday   
4   295      Brian     Bryon  Social Media  Python For Data Science  Weekend   

  GENDER  
0   Male  
1   Male  
2   Male  
3   Male  
4   Male

In [56]:
#no of categoriess in each column
print("No of cat in: ")

for col in enr_df[["SOURCE","COURSE","TRACK","GENDER"]]:
    print("{} = {}".format(col, len(enr_df[col].unique())))
    

No of cat in: 
SOURCE = 6
COURSE = 16
TRACK = 5
GENDER = 2


In [57]:
enr_df.isnull().sum()

ID             0
FIRST NAME     0
LAST NAME      0
SOURCE        12
COURSE        25
TRACK          0
GENDER         0
dtype: int64

In [59]:
#check for the frequently occuring
enr_df["SOURCE"].value_counts()

Social Media    1096
WEBSITE          423
Mr Simps         222
Male              12
Female             1
Name: SOURCE, dtype: int64

In [60]:
#fill with frequently occuring
enr_df['SOURCE'].replace(np.nan, "Social Media", inplace=True)

In [61]:
#check if filled
enr_df.isnull().sum()

ID             0
FIRST NAME     0
LAST NAME      0
SOURCE         0
COURSE        25
TRACK          0
GENDER         0
dtype: int64

In [62]:
#check for the frequently occuring
enr_df["COURSE"].value_counts()

Full Stack                      299
FullStack                       298
Product Design(UI/UX)           123
Figma Design to Webflow         122
Frontend Web Development        121
Product Design(UI/UX) ONLINE    112
Python For DataScience          102
Product Design                  100
CyberSecurity                    97
Product Management               93
Backend With Python Django       92
Frontend                         88
Python For Data Science          81
Male                             12
Female                            1
Name: COURSE, dtype: int64

In [71]:
reg_df.head(5)

ID FIRST NAME LAST NAME        SOURCE                   COURSE    TRACK  \
0  2373      Brian       May  Social Media   Python For DataScience  Weekend   
1   758       Leon    Melvin  Social Media    Product Design(UI/UX)  Weekend   
2  2287       Juan    Harris      Mr Simps    Product Design(UI/UX)  Weekday   
3  2480    Gilbert    Denson  Social Media  Figma Design to Webflow  Weekend   
4  1359      Larry  Williams       WEBSITE   Python For DataScience  Weekday   

  GENDER  
0   Male  
1   Male  
2   Male  
3   Male  
4   Male

In [74]:
#group by gender and count
reg_df.groupby(["GENDER"]).count()


ID  FIRST NAME  LAST NAME  SOURCE  COURSE  TRACK
GENDER                                                       
Female      717         717        717     717     717    717
Male       2903        2903       2903    2877    2878   2903
No_gender    56          56         56      56      56     56

In [76]:
enr_df.groupby(["GENDER"]).count()

ID  FIRST NAME  LAST NAME  SOURCE  COURSE  TRACK
GENDER                                                    
Female   143         143        143     143     143    143
Male    1623        1623       1623    1623    1598   1623

# probability that a randomly selected registered female will be enrolled in the program

In [84]:
no_female_registered = (reg_df['GENDER'] == "Female").sum()
total_registerd = len(reg_df["GENDER"])
#no_female_registered
#total_registerd
#prob_reg_female_enr = 
no_female_enrolled = (enr_df['GENDER'] == "Female").sum()
prob_enr_female = (no_female_enrolled/no_female_registered)*100
prob_enr_female


19.9442119944212

### The probabilty that a randomly selected registered female will be enrolled is 19.94 percent

# Probabilty that a randomly selected male will be enrolled in the program

In [85]:
no_male_registered = (reg_df['GENDER'] == "Male").sum()

no_male_enrolled = (enr_df['GENDER'] == "Male").sum()
prob_enr_male = (no_male_enrolled/no_male_registered)*100
prob_enr_male

55.90768170857734

### The probabilty that a randomly selected registered male will be enrolled is 55.91 percent

# Probability that a registerd applicant will eventually enroll in the program

In [86]:
total_registerd = len(reg_df["GENDER"])
total_enrolled = len(enr_df["GENDER"])

prob_of_enrollement = (total_enrolled/total_registerd)*100
prob_of_enrollement

48.04134929270947

#### The probabilty that a randomly selected registered applicant will be enrolled is 48.04 percent

# Full probability distribution for the genders